config/fcos/v1_5_exp/translate.go

```diff
@@ -68,17 +68,31 @@ func (c Config) ToIgn3_4Unvalidated(options common.TranslateOptions) (types.Conf
 		return types.Config{}, translate.TranslationSet{}, r
 	}
 	r.Merge(c.processBootDevice(&ret, &ts, options))
+
 	for i, disk := range ret.Storage.Disks {
-		// In the boot_device.mirror case, nothing specifies partition numbers
-		// so match existing partitions only when `wipeTable` is false
-		if !util.IsTrue(disk.WipeTable) {
-			for j, partition := range disk.Partitions {
-				// check for reserved partlabels
-				if partition.Label != nil {
+		for p, partition := range disk.Partitions {
+			if partition.Label != nil {
+				if *partition.Label == "root" {
+					if partition.SizeMiB != nil && *partition.SizeMiB < 8192 {
+						r.AddOnWarn(path.New("json", "storage", "disks", i, "partitions", p, "sizeMiB", *partition.SizeMiB), common.ErrRootTooSmall)
+					} else {
+						for _, op := range disk.Partitions {
+							if op.Number > partition.Number {
```

Comment:
We're currently performing this check for every partition after the root partition, not just the one immediately after it.

Reply:
OH can we assume that they are always in order?

Reply:
What we don't want is to warn repeatedly for every additional partition.  But you're right, we can't assume they're in order, either by starting offset or by partition number.  And actually, we can't make assumptions about `op.Number` either, since it might be 0.

So we'll need an heuristic for figuring out the earliest partition start.  If the `startMiB` is always specified, this is easy.  Otherwise, the partition order is determined by a combination of `sgdisk`'s heuristics (see the man page for the `-n` option) and the additional heuristics applied by [partitions.go](https://github.com/coreos/ignition/blob/main/internal/exec/stages/disks/partitions.go).  And unfortunately, they have some information we don't, which is the overall size of the disk, and thus the size of the largest available block.

So we'll need to figure this out.  If there are cases where we're not sure, it's better to not warn.